shell/browser/api/electron_api_session.cc

```diff
@@ -294,6 +298,10 @@ Session::~Session() {
     service->SetHunspellObserver(nullptr);
   }
 #endif
+
+#if BUILDFLAG(ENABLE_ELECTRON_EXTENSIONS)
+  extensions::ExtensionRegistry::Get(browser_context())->RemoveObserver(this);
```

Comment:
I think this is handled by `ScopedObserver`, yes?

Given we're only ever only ever observing a single source, perhaps the scoped observer isn't needed?

Reply:
The `ScopedObserver` wasn't really working how I thought it did so I just removed it. It would need to be passed the `browser_context()` which the template doesn't support.